test/widget_tests/splash_screen_test.dart

```diff
@@ -0,0 +1,66 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:movies_app/presentation/view/main_screen.dart';
+import 'package:movies_app/presentation/view/splash_screen.dart';
+import 'package:animated_splash_screen/animated_splash_screen.dart';
+import 'package:mockito/annotations.dart';
+import 'splash_screen_test.mocks.dart';
+
+@GenerateNiceMocks(
+  [
+    MockSpec<NavigatorObserver>(),
+  ],
+)
+void main() {
+  late MockNavigatorObserver mockNavObserver;
+
+  setUp(() {
+    mockNavObserver = MockNavigatorObserver();
+  });
+
+  Widget buildWidget() {
+    return MaterialApp(
+      home: const SplashScreen(
+        key: Key('screenKey'),
```

Comment:
screenKey? This key i think is more related to Splash

Reply:
screenKey is not used so it will be removed. Done